### PR TITLE
Feature/stage images

### PR DESCRIPTION
**Pull request by PabloPie**
Poc pulls images from registry in case of staging deployment, builds from source in case of dev deployment.

### CODE REVIEW

File: hosts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [caliopen:vars]
-use_https=false
+use_https=true
+stage=true
 vhost_name=poc.caliopen.org
+compose_file="{{ '/opt/caliopen/devtools/docker-compose.staging.yml' if stage == 'true' else '/opt/caliopen/devtools/docker-compose.yml' }}"
```
Comment (PabloPie) [inline hosts]: Maybe there is a better way to make this variable available for both the main playbook and the load data one.

File: roles/caliopen/tasks/main.yaml [hunk 2]
```diff
@@ -94,3 +102,4 @@
 
 - name: start full caliopen docker stack
   service: name=caliopen-stack state=started
+  notify: restart nginx
```
Comment (PabloPie) [inline roles/caliopen/tasks/main.yaml:105]: nginx needs to be restarted after the frontend has started

**Comment by PabloPie** [comment]
Note that this only works with the update to the docker-compose files in Caliopen/Caliopen@a31c7586c7935f84c2367094f71cb954d7aa5963.

File: hosts (hunk 1 above)
Comment (iamdey) [reply hosts]: I've no idea ^^

**Review by iamdey — APPROVED** [review]
LGTM

File: roles/caliopen/tasks/main.yaml [hunk 3]
```diff
@@ -83,9 +84,16 @@
     host: 0.0.0.0
     port: 9042
     delay: 30
+  when: stage == 'false'
+
+- name: pull docker services images
```
Comment (PabloPie) [inline roles/caliopen/tasks/main.yaml:89]: Waiting for cassandra to be up on port 9042 is not a option for staging, atm we just pull the images to make some time before starting the setup. Not the best option.

**Comment by iamdey** [comment]
it works almost great!

the task witch installing a brand new instance works without failure.

But when we try to load fixtures it fails like that:

```
bin/load-data.sh 
SUDO password: 

PLAY [caliopen] *************************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************************
ok: [docker-stack]

TASK [Create a Caliopen Admin] **********************************************************************************************************************************************************
fatal: [docker-stack]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'devtools_dir' is undefined\n\nThe error appears to have been in '/home/dey/Workspace/caliopen/ansible-poc/playbook-load-data.yml': line 7, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n  - name: Create a Caliopen Admin\n    ^ here\n"}
```

A solution is to move `devtools_dir=/opt/caliopen/devtools/` into hosts file

**Comment by PabloPie** [comment]
I think there is a better way of doing it, instead of listing variables in the inventory file we need to define the 'all' role, variables part of that role can be read by every other role. That way when you do dynamic inventory you don't lose your variables either.

cc @gdchamal

**Review by gdchamal — APPROVED** [review]

**Comment by PabloPie** [comment]
@iamdey Had the same error when loading fixtures, I moved the variable to the hosts file.

**Pull request merged**